File: ebs/src/main/kotlin/com/bft/ebs/services/impl/BookService.kt
package com.bft.ebs.services.impl

import com.bft.ebs.dto.BookDto
import com.bft.ebs.repositories.impl.BookRepository
import com.bft.ebs.services.EbsMapper
import com.bft.ebs.services.interfaces.BookService
import org.springframework.stereotype.Service

@Service
class BookService(private val bookRepository: BookRepository) :
    BookService {
    override fun getBook(id: Long): BookDto {
        return EbsMapper.mapBookToBookDto(bookRepository.read(id))
    }

    override fun getAllBooks(): List<BookDto> {
        return bookRepository.findAll().map { EbsMapper.mapBookToBookDto(it) }
    }

    override fun createBook(bookDto: BookDto): Long {
        return bookRepository.create(EbsMapper.mapBookDtoToBook(bookDto)).id
            ?: throw IllegalStateException("Book already exists")
    }

    override fun deleteBook(id: Long): Boolean {
        return bookRepository.delete(id)
    }

    override fun updateBook(bookDto: BookDto): Boolean {
        return bookRepository.update(EbsMapper.mapBookDtoToBook(bookDto))
    }
}

File: front/ebs-frontend/src/services/BookService.js
import axios from "axios";
import { REST_API_BASE_URL } from '../../config'

const SERVICE_PATH = REST_API_BASE_URL + 'books/'

export const listBooks = () => axios.get(SERVICE_PATH + 'all');

export const createBook = (book) => axios.post(SERVICE_PATH + 'create', book)

export const getBook = (bookId) => axios.get(SERVICE_PATH + 'get/' + bookId)

export const updateBook = (bookId, book) => axios.put(SERVICE_PATH + 'update/' + bookId, book)

export const deleteBook = (bookId) => axios.delete(SERVICE_PATH + 'delete/' + bookId)